crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -138,7 +140,17 @@ impl<'a> Resolver<'a> {
             let name = self.interner.definition_name(unused_var.id);
             if name != ERROR_IDENT {
                 let ident = Ident(Spanned::from(unused_var.location.span, name.to_owned()));
-                self.push_err(ResolverError::UnusedVariable { ident });
+                let mut const_found = false;
+                for (_, const_info) in self.interner.get_all_global_consts() {
+                    if const_info.ident == ident {
```

Comment:
I'd rather issue an unused error than have to do a linear search through all consts in the program on each unused variable error. E.g. `unused_var` should have a is_global flag we can check directly, or remove the check entirely

Reply:
I agree, I'm changed it to use a flag which is much better. I don't want to just issue an unused err though as a const that is used in one function but unused in another will still result in an unused error.